app/controllers/products_controller.rb

```diff
@@ -0,0 +1,51 @@
+class ProductsController < ApplicationController
+
+  def new
+    @product = Product.new
+    @vendor = Vendor.find(params["format"])
+  end
+
+  def create
+    @product = Product.new(product_edit_params[:product])
+    if @product.save
+      redirect_to vendor_path(@product.vendor_id)
+    else
+      @vendor = @product.vendor
+      render :new
+    end
+  end
+
+  def index
+    @products = Product.all
+  end
+
+  def destroy
+    @product = Product.find(params[:id])
+    if @product.destroy
+      redirect_to vendor_path(@product.vendor_id)
+    else
+      flash.now[:alert] = 'Product could not be deleted.'
+    end
+  end
+
+  def edit
+    @product = Product.find(params[:id])
+  end
+
+  def update
+    @product = Product.find(params[:id])
+    if @product.update(product_edit_params[:product])
+      redirect_to vendor_path(@product.vendor_id)
+    else
+      @vendor = @product.vendor
+      render :edit
+    end
```

Comment:
I am not sure how you are using the @vendor instance variable here. It looks like your edit method just needs the @product, and @vendor does not appear to be used in the view.